code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm

```diff
@@ -586,14 +586,14 @@ GLOBAL_LIST_EMPTY_TYPED(air_alarms, /obj/machinery/airalarm)
 
 	selected_mode.replace(my_area, pressure)
 
-/obj/machinery/airalarm/proc/select_mode(atom/source, datum/air_alarm_mode/mode_path)
+/obj/machinery/airalarm/proc/select_mode(atom/source, datum/air_alarm_mode/mode_path, should_apply = TRUE) //BUBBERSTATION CHANGE: ADDS should_apply
 	var/datum/air_alarm_mode/new_mode = GLOB.air_alarm_modes[mode_path]
 	if(!new_mode)
 		return
 	if(new_mode.emag && !(obj_flags & EMAGGED))
 		return
 	selected_mode = new_mode
-	selected_mode.apply(my_area)
+	if(should_apply) selected_mode.apply(my_area) //BUBBERSTATION CHANGE: ADDS should_apply
```

Comment:
yes it's valid but this ain't now 2013 rodeo pal

Reply:
what do you mean by this